applications/FluidDynamicsBiomedicalApplication/custom_utilities/parabolic_profile_utilities.cpp

```diff
@@ -0,0 +1,241 @@
+        }
+
+        // Get the new maximum distance
+        current_max_dist = block_for_each<MaxReduction<double>>(rFluidModelPart.Nodes(), [](const NodeType& rNode){
+            return rNode.GetValue(WALL_DISTANCE);
+        });
+        rFluidModelPart.GetCommunicator().GetDataCommunicator().MaxAll(current_max_dist);
+    }
+
+    // Reset boundary values to zero
+    block_for_each(rWallModelPart.Nodes(), [](NodeType& rNode){
+        rNode.GetValue(WALL_DISTANCE) = 0.0;
+    });
+}
+
+void ParabolicProfileUtilities::ImposeParabolicInlet(
+    ModelPart& rModelPart,
+    const double MaxParabolaValue)
+{
+    // Impose the parabolic inlet profile
+    ImposeParabolicProfile(rModelPart, MaxParabolaValue);
+}
+
+void ParabolicProfileUtilities::ImposeParabolicInlet(
+    ModelPart& rModelPart,
+    const GenericFunctionUtility::Pointer MaxParabolaValue)
+{
+    // Impose the parabolic inlet profile
+    ImposeParabolicProfile(rModelPart, MaxParabolaValue);
+}
+
+template<class TInputType>
+void ParabolicProfileUtilities::ImposeParabolicProfile(
+    ModelPart &rModelPart,
+    const TInputType& rMaxParabolaValue)
+{
+    // Get time value from model part ProcessInfo
+    const auto& r_process_info = rModelPart.GetProcessInfo();
+    KRATOS_ERROR_IF_NOT(r_process_info.Has(TIME)) << "TIME is not present in '" << rModelPart.FullName() << "' ProcessInfo." << std::endl;
+    const double time = r_process_info[TIME];
+
+    // Get the maximum distance to the wall
+    const double max_dist = block_for_each<MaxReduction<double>>(rModelPart.Nodes(), [](NodeType& rNode){
+        return rNode.GetValue(WALL_DISTANCE);
+    });
+    KRATOS_ERROR_IF(std::abs(max_dist) < 1.0e-12) << "WALL_DISTANCE is close to zero." << std::endl;
+
+    // Impose the parabolic profile values
+    block_for_each(rModelPart.Nodes(), array_1d<double,3>(), [time, max_dist, &rMaxParabolaValue](NodeType& rNode, array_1d<double,3>& rTLSValue){
+        //Calculate distance for each node
+        const double wall_dist = rNode.GetValue(WALL_DISTANCE) < 0.0 ? 0.0 : rNode.GetValue(WALL_DISTANCE);
+
+        // Calculate the inlet direction
+        rTLSValue = rNode.GetValue(INLET_NORMAL);
+        const double n_norm = norm_2(rTLSValue);
+        if (n_norm > 1.0e-12) {
+            rTLSValue /= -n_norm;
+        } else {
+            KRATOS_WARNING("ImposeParabolicProfile") << "Node " << rNode.Id() << " INLET_NORMAL is close to zero." << std::endl;
+            rTLSValue /= -1.0;
+        }
+
+        // Calculate the inlet value module
+        const double max_value = GetMaxParabolaValue(time, rNode, rMaxParabolaValue);
+        const double value_in = max_value * (1.0-(std::pow(max_dist-wall_dist,2)/std::pow(max_dist,2)));
+        rTLSValue *= value_in;
+
+        // Set and fix the VELOCITY DOFs
+        rNode.FastGetSolutionStepValue(VELOCITY) = rTLSValue;
+        rNode.Fix(VELOCITY_X);
+        rNode.Fix(VELOCITY_Y);
+        rNode.Fix(VELOCITY_Z);
```

Comment:
Is it necessary to fix always when this transient value is computed? Wouldn't it be better if fixing of variables are moved out and only called once?

Reply:
No it needs to be fixed and unfixed at each time step as we normally do in all processes. The point of doing it in here is that I avoid looping the nodes twice, one to assign the value and one to fix it.